src/hera/task.py

```diff
@@ -341,7 +342,14 @@ def get_parameters(self) -> List[V1alpha1Parameter]:
             args = set(inspect.getfullargspec(self.func).args).intersection(set(self.input_from.parameters))
             for arg in args:
                 parameters.append(V1alpha1Parameter(name=arg, value=f'{{{{item.{arg}}}}}'))
+        if self.func:
+            parameters += self._get_func_parameters()
+
+        return parameters
 
+    def _get_func_parameters(self):
```

Comment:
Similarly to above, I think this can be public 🙂

Reply:
Addressing this and the one above, wondering what the justification would be to make this public? Outside of the task class what benefit would there be to access these?

Reply:
The main reason is exposing parts of Hera's internals in order to offer some flexibility to users who build their own `Task` on top of the Hera one. For instance, if there are users who want to inject something in their template script, in addition to what Hera injects already, they can access the public method to obtain the pre-built one, and modify that one, rather than write the script assemble method themselves. This kind of applies generally - users can use any of the public functions to use the "nice" aspect of Hera, which sets up a bunch of stuff already, so users save some work and only have to be concerned with "what do _I_ want to inject further into Argo object definitions that helps my problem".

Reply:
My concern with this would be that it is introducing a certain level of brittleness to the code base, for example, if at a certain point we needed to change the method signature, we would be introducing a breaking change.
Perhaps, for the getter `_get_func_parameters` I hear, but for the validate method, would it not be `validate` is the public method, and than internal calls are private?

Reply:
I definitely see where you're coming from. Perhaps the balance between a public `get_func_parameters` and a private `validate` that you suggested works best for now. So, let's go with your suggestion! 🙂